mattermost/mattermost-mobile
# Fix gallery footer author image showing old image

#### Summary
The gallery avatar was not taking into consideration the LastPictureUpdate, so you may end up seeing an old cached picture instead of the correct profile picture of the person.

I also piggybacked some refactoring so all avatars do the same process to get the picture.

#### Ticket Link
Fix https://mattermost.atlassian.net/browse/MM-57887

#### Test steps
With user A, post a picture in a channel
With user B, on a phone, confirm the profile picture next to the post is the same as the one in the gallery footer when opening the picture.
With user A, change the profile picture
With user B, on a phone, confirm the profile picture have changed both in the avatar next to the post, and in the gallery footer.
 
#### Release Note
```release-note
Fix gallery footer not showing an old profile picture
```
